src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -396,16 +424,186 @@ private void buildSectionPanelsForRecipientQuestionGiver(
+                    
+                    // setup for next section
+                    teamsWithResponses.clear();
+                    teamsWithResponses.add(currentTeam);
+                    
+                    sectionPanel = new InstructorFeedbackResultsSectionPanel();
+                }
+                sectionPanel.getIsTeamWithResponses().put(currentTeam, true);
+            }
+            
+            // Build recipient panel for the current response's recipient
+            InstructorFeedbackResultsGroupByParticipantPanel recipientPanel 
+                    = buildGroupByParticipantPanel(recipientIdentifier, currentTeam, additionalInfoId, recipientIndex,
+                                                   recipientToGiverToResponsesMap);
+            addParticipantPanelToSectionPanel(sectionPanel, currentTeam, recipientPanel);
+            
+            teamMembersWithResponses.add(recipientIdentifier);
+
+            prevTeam = currentTeam;
+            prevSection = currentSection;
+        }
+        
+        // for the last section with responses
+        buildMissingParticipantPanelsForTeam(sectionPanel, prevTeam, teamMembersWithResponses, viewType.isFirstGroupedByGiver());
+        
+        teamsWithResponses.add(prevTeam);
+        buildMissingTeamAndParticipantPanelsForSection(sectionPanel, prevSection, teamsWithResponses, viewType.isFirstGroupedByGiver());
+        
+        finaliseBuildingSectionPanelWithoutStats(sectionPanel, prevSection);
+        sectionPanels.put(prevSection, sectionPanel);   
+        if (selectedSection.equals("All")) {
+            sectionsWithResponses.add(prevSection); // for the last section having responses 
+            buildSectionPanelsForMissingSections(sectionsWithResponses);
+        }
+    }
+
+    private InstructorFeedbackResultsGroupByParticipantPanel buildGroupByParticipantPanel(
+                                    String primaryParticipantIdentifier, String currentTeam,
+                                    String additionalInfoId, int primaryParticipantIndex,
+                                    Map.Entry<String, Map<String, List<FeedbackResponseAttributes>>> primaryToSecondaryToResponsesMap) {
+        // first build secondary participant panels for the primary participant panel
+        Map<String, List<FeedbackResponseAttributes>> giverToResponsesMap = primaryToSecondaryToResponsesMap.getValue();
+        List<InstructorFeedbackResultsSecondaryParticipantPanelBody> secondaryParticipantPanels 
+                             = buildSecondaryParticipantPanels(
+                                        additionalInfoId, primaryParticipantIndex, giverToResponsesMap);
+        
+        // construct the primary participant panel
+        String primaryParticipantNameWithTeamNameAppended = bundle.appendTeamNameToName(bundle.getNameForEmail(primaryParticipantIdentifier), bundle.getTeamNameForEmail(primaryParticipantIdentifier));
```

Comment:
Either use temporary variables or line breaks, this line is over 200 characters now.